### PR TITLE
[docs] [ENG-8973] Update docs re submit profiles

### PR DESCRIPTION
**Pull request by radoslawkrzemien**
# Why

https://linear.app/expo/issue/ENG-8973/eas-submit-doesnt-use-asc-configuration-from-selected-build-profile

# How

Updated docs under https://docs.expo.dev/submit/eas-json/ to include the description of new changes regarding how the submission profile is resolved when the user does not provide the profile name

# Test Plan

Checked manually
Screenshot 2023-10-31 at 12 13 18

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/submit/eas-json.mdx [hunk 1]
```diff
@@ -71,7 +71,9 @@ The JSON object under `submit` can contain multiple submit profiles. Each profil
 }
 ```
 
-You can use EAS CLI to pick up another `submit` profile by specifying it with a parameter. For example, `eas submit --platform iOS --profile submit-profile-name`.
+When you select a build for submission, then a profile matching the build profile used for the selected build will be chosen. If it does not exist, it will default to `production` profile.
```
Comment (amandeepmittal) [inline docs/pages/submit/eas-json.mdx:74]:
```suggestion
When you select a build for submission, it chooses the profile that is used for the selected build. If the profile does not exist, it selects the default `production` profile.
```

Let's use active voice here for better readability.

**Review by amandeepmittal — APPROVED** [review]
One tine suggestion but LGTM 👍 Thank you for updating this!

File: docs/pages/submit/eas-json.mdx [hunk 2]
```diff
@@ -71,7 +71,9 @@ The JSON object under `submit` can contain multiple submit profiles. Each profil
...
+
+You can also use EAS CLI to pick up another `submit` profile by specifying it with a parameter. For example, `eas submit --platform iOS --profile submit-profile-name`.
```
Comment (szdziedzic) [inline docs/pages/submit/eas-json.mdx:76]:
```suggestion
You can also use EAS CLI to pick up another `submit` profile by specifying it with a parameter. For example, `eas submit --platform ios --profile submit-profile-name`.
```

**Review by szdziedzic — APPROVED** [review]

**Review by Simek — CHANGES_REQUESTED** [review]
Can you revert the lock changes? Looks like you have used newer version of Yarn locally than the docs uses by default.

**Review by Simek — APPROVED** [review]

**Pull request merged**